core/util.bzl

```diff
@@ -47,6 +47,70 @@ def cp(repository_ctx, src, dest = None):
 
     return dest
 
+def cp_many(repository_ctx, srcs):
+    """Copy the given files into the external repository root.
+
+    If possible this is done by invoking tar from an external shell process.
+    This should be more efficient than writing files with repository_ctx.file
+    and preserving their permissions by invoking chmod on every file.
+
+    Args:
+      repository_ctx: The repository context of the current repository rule.
+      srcs: A list of source files to copy.
+
+    Returns:
+      A list of destination paths for the files copied into the repository.
+    """
+    tar_list = []
+    repository_files = []
+
+    # Tar will run with outputBase as its working directory.
+    repository_dir = repository_ctx.execute(["pwd"]).stdout.strip("\n")
+    output_base = repository_dir.removesuffix("external/" + repository_ctx.name)
+
+    # Fallback to cp if outputBase cannot be determined.
+    if output_base == repository_dir:
+        return [cp(repository_ctx, src) for src in srcs]
+
+    for src in srcs:
+        abs_path = str(repository_ctx.path(src))
+        rel_path = abs_path.removeprefix(output_base)
```

Comment:
[Bazel Skylib](https://github.com/bazelbuild/bazel-skylib/blob/872e9b06e18ae8ba2897cb65b9aaa172aa6279f3/lib/paths.bzl#L15) offers a file filepath mangling functions that could be better suited here.

Reply:
Good suggestion - I'll take a look :+1: